cognite/client/data_classes/transformation_jobs.py

```diff
@@ -0,0 +1,249 @@
+        destination_database (str): Target database if the destination type is raw.
+        destination_table (str): Target table name if the destination type is RAW.
+        conflict_mode (str): What to do in case of id collisions: either "abort", "upsert", "update" or "delete".
+        raw_query (str): Query of the transformation that is being executed.
+        error (str): Error message from the server.
+        ignore_null_fields (bool): Indicates how null values are handled on updates: ignore or set null.
+        created_time (int): Time when the job was created.
+        started_time (int): Time when the job started running.
+        finished_time (int): Time when the job finished running.
+        last_seen_time (int): Time of the last status update from the job.
+        cognite_client (CogniteClient): The client to associate with this object.
+    """
+
+    def __init__(
+        self,
+        id: int = None,
+        uuid: UUID = None,
+        status: TransformationJobStatus = None,
+        transformation_id: int = None,
+        source_project: str = None,
+        destination_project: str = None,
+        destination: "TransformationDestination" = None,
+        conflict_mode: str = None,
+        raw_query: str = None,
+        error: str = None,
+        ignore_null_fields: bool = False,
+        created_time: int = None,
+        started_time: int = None,
+        finished_time: int = None,
+        last_seen_time: int = None,
+        cognite_client=None,
+    ):
+        self.id = id
+        self.uuid = uuid
+        self.status = status
+        self.transformation_id = transformation_id
+        self.source_project = source_project
+        self.destination_project = destination_project
+        self.destination = destination
+        self.conflict_mode = conflict_mode
+        self.raw_query = raw_query
+        self.error = error
+        self.ignore_null_fields = ignore_null_fields
+        self.created_time = created_time
+        self.started_time = started_time
+        self.finished_time = finished_time
+        self.last_seen_time = last_seen_time
+        self._cognite_client = cognite_client
+
+    def update(self):
+        """`Get updated job status. <https://docs.cognite.com/api/playground/#operation/runTransformation>`_"""
+        updated = self._cognite_client.transformations.jobs.retrieve(id=self.id)
+        self.status = updated.status
+        self.error = updated.error
+        self.started_time = updated.started_time
+        self.finished_time = updated.finished_time
+        self.last_seen_time = updated.last_seen_time
+
+    def metrics(self):
+        """`Get job metrics. <https://docs.cognite.com/api/playground/#operation/runTransformation>`_"""
+        return self._cognite_client.transformations.jobs.list_metrics(self.id)
+
+    def wait(self, polling_interval: float = 1, timeout: Optional[float] = None) -> "TransformationJob":
```

Comment:
these should belong to api client maybe?

Reply:
`metrics()` is a shortcut to enable more readable use of the sdk, same as `run()` in `transformation` data class:

This two blocks do the same, start running a transformation, and retrieve the metrics when it finishes:

```
my_metrics = my_transformation.run().metrics()
```

```
my_job = client.transformations.jobs.run(id = my_transformation.id)
my_metrics = client.transformations.jobs.list_metrics(id = my_job.id)
```

we decided to use a callable method instead of a property, to make clear to the user that an expensive operation (an API call) is performed on that "navigation".

Reply:
`wait()`, `wait_async()` and `update()` were done as instance methods because they don't map directly to an API operation, but to operations involving local and server operations, to achieve something the user might want to actually do with a `TransformationJob` instance, e.g. the user might want to `update()` the status of its instance, instead of creating a new object (this calls the api and maps the received fields to the calling object), or to `wait()` blocking the current thread until the transformation finishes running on the server.